Adds example request and Json parameter information for container

draghuram opened the pull request:
start API.

Closes #10304.

Signed-off-by: Raghuram Devarakonda draghuram@gmail.com

draghuram committed:
Adds example request and Json parameter information for container
start API.

Closes #10304.

Signed-off-by: Raghuram Devarakonda <draghuram@gmail.com>

cpuguy83 commented:
LGTM

moxiegirl commented:
LGTM

moxiegirl commented:
ping @jamtur01 @fredlf @thaJeztah (for those eagle eyes maintainer or no)

thaJeztah commented on docs/sources/reference/api/docker_remote_api_v1.18.md:
```diff
@@ -676,12 +676,90 @@ Start the container `id`
+        a host path into the container), or `host_path:container_path:ro`
+        (to make the bind-mount read-only inside the container).
+-   **Links** - A list of links for the container.  Each link entry should be of
+      of the form "container_name:alias".
```
"container_name:alias" should be in back-ticks; `"container_name:alias"`

thaJeztah commented on docs/sources/reference/api/docker_remote_api_v1.18.md:
```diff
@@ -676,12 +676,90 @@ Start the container `id`
+        (to make the bind-mount read-only inside the container).
+-   **Links** - A list of links for the container.  Each link entry should be of
+      of the form "container_name:alias".
+-   **LxcConf** - LXC specific configurations.  These configurations will only
```
Looks like *two* spaces before "These"

thaJeztah commented on docs/sources/reference/api/docker_remote_api_v1.18.md:
```diff
@@ -676,12 +676,90 @@ Start the container `id`
+      Take note that `port` is specified as a string and not an integer value.
+-   **PublishAllPorts** - Allocates a random host port for all of a container's
+      exposed ports. Specified as a boolean value.
+-   **Privileged** - Gives the container full access to the host.  Specified as
```
Looks like *two* spaces before "Specified"

thaJeztah commented on docs/sources/reference/api/docker_remote_api_v1.18.md:
```diff
@@ -676,12 +676,90 @@ Start the container `id`
+-   **Capdrop** - A list of kernel capabilties to drop from the container.
+-   **RestartPolicy** – The behavior to apply when the container exits.  The
+        value is an object with a `Name` property of either `"always"` to
+        always restart or `"on-failure"` to restart only when the container
```
Is "none" also a valid option here? Or is that `""` in the API?

thaJeztah replied:
^^ I think @cpuguy83 will know the answer to this

cpuguy83 replied:
Hmm, it looks like it could literally be anything, but only `always` and `on-failure` do anything.

thaJeztah commented on docs/sources/reference/api/docker_remote_api_v1.18.md:
```diff
@@ -676,12 +676,90 @@ Start the container `id`
+      `Ulimits: { "Name": "nofile", "Soft": 1024, "Hard", 2048 }}`
+-   **SecurityOpt**: A list of string values to customize labels for MLS
+    systems, such as SELinux.
+-   **LogConfig** - Logging configuration to container, format
```
Weird sentence: "Logging configuration *to* container"

fredlf replied:
Agreed. Should it be "Log configuration for the container" ?

thaJeztah commented on docs/sources/reference/api/docker_remote_api_v1.18.md:
```diff
@@ -676,12 +676,90 @@ Start the container `id`
+-   **SecurityOpt**: A list of string values to customize labels for MLS
+    systems, such as SELinux.
+-   **LogConfig** - Logging configuration to container, format
+      `{ "Type": "<driver_name>", "Config": {"key1": "val1"}}
```
Missing closing back-tick. I actually think this was fixed in https://github.com/docker/docker/pull/12205 (perhaps this needs a rebase?)

thaJeztah replied:
Oh, never mind, https://github.com/docker/docker/pull/12205 is fixing that in a different section of this document

thaJeztah commented:
Added some notes in the 1.18 document; I think the 1.19 document changes are a copy of that, so I didn't put comments there.

Also; as I think this information is copied from *existing* docs, we may need a follow-up PR to correct these issues I'm the other documents and sections.

Thanks @draghuram!

fredlf commented:
I agree with the eagle-eyed copy edits of @thaJeztah, once those are fixed, LGTM.

draghuram commented:
I will fix the changes and update the PR.

draghuram commented:
@thaJeztah Amazingly detailed review :-). 

I have made the following changes:

- Added missing back ticks at couple of places.
- Removed double spacing before the word "These". I also found couple of other cases similar to this and fixed those as well. 
- "LogConfig" now reads as "Log configuration for the container, specified as". 
- I noticed that "HostConfig" section in "Create a Container" API is not properly indented and fixed it. 

I followed the instructions at https://docs.docker.com/project/review-pr/ and pushed the branch after doing `git commit --amend`. I had to force push since `git commit --amend` replaces existing commit with a new one. I am guessing the force push is the reason for "merge conflicts" as reported above. Perhaps, I should have pushed my later changes as a second independent commit? If so, the contributor guide needs to be fixed accordingly. 

In any case, I can submit the changes as a separate PR if that makes the process simpler in this case.

thaJeztah commented on docs/sources/reference/api/docker_remote_api_v1.18.md:
```diff
@@ -212,56 +212,56 @@ Json Parameters:
-        Available types: `json-file`, `syslog`, `none`.
-        `json-file` logging driver.
-  -   **CgroupParent** - Path to cgroups under which the cgroup for the container will be created. If the path is not absolute, the path is considered to be relative to the cgroups path of the init process. Cgroups will be created if they do not already exist.
+    -   **Binds** – A list of volume bindings for this container. Each volume
```
Note to reviewers; these whitespace changes look correct; this and below is a nested bullet-list for HostConfig

thaJeztah commented:
Thanks, @draghuram ! I think the merge error means you will have to "rebase" your PR (and fix possible merge conflicts). You can find instructions here: https://docs.docker.com/project/work-issue/#pull-and-rebase-frequently

If you need assistance with that, don't hesitate to ask for help in the #docker-dev IRC channel.

For reviewers, use this link to view the changes, ignoring the white-space changes; https://github.com/docker/docker/pull/12218/files?w=1

draghuram commented:
I updated the PR after rebasing.

thaJeztah commented:
"all green" now, great :)
